ruffle: remove unneeded dependencies

normalcea opened the pull request:
This PR removes unneeded dependencies for ruffle's desktop version. As of this [commit](https://github.com/ruffle-rs/ruffle/commit/c12130a5940fc55fb41b2089a4857f58ec2ed07d) in ruffle, it's more clear that ruffle only requires `alsa-lib` and `udev` development libraries as `buildInputs` for Linux.

The `postInstall` phase as been adjusted accordingly with `makeBinaryWrapper` replacing `wrapGAppsHook3`

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
ruffle: refactor to correct dependencies

jchv commented:
This fails to load over XWayland:

```
Error: os error at /build/ruffle-0-nightly-2025-05-11-vendor/winit-0.30.9/src/platform_impl/linux/mod.rs:787: Failed to load one of xlib's shared libraries
```

(I'd like to add a NixOS test for this at some point, but oh well.)

I'm also wondering if we can tell with higher certainty that we really don't need the gapps wrapper. I'm running on KDE with the KDE portal so e.g. the file dialog will obviously work just fine without GLib schemas, but there may be a case (say, on an environment like SwayWM, where you won't have env pollution) where some library (a message box library?) tries to invoke GTK and runs afoul of GLib schema issues. Unfortunately, it's quite a PITA to actually find situations like this, especially since env pollution commonly makes things look like they are working when they are not.

normalcea commented:
I looked at `pkgs/by-name/bo/boilr/package.nix` and I found the list of xorg packages that complete the winit x11 backend. I tested this with GNOME under xorg and it launches properly.

jchv commented:
> By the way how did you force an application to run under xwayland?

The most obvious way is to just clear the `WAYLAND_DISPLAY` env var, e.g.

```
WAYLAND_DISPLAY= ruffle
```

> On the Gapps wrapper I think it would just be handled by the desktop portal the compositor uses (which I think is `xdg-desktop-portal-gtk` for sway don't quote me on that) and I'm not sure how glib schemas would play with that. I'd like to do further testing in non-GNOME environments from now on. Ruffle on GNOME also uses the correct `xdg-desktop-portal-gnome` (libadwaita file picker) instead of the gtk file picker if you were wondering.

I think the file chooser will always go through portals, so that part should be safe since it will be out-of-process. If we don't have any known cases where we run into this then I'm fine dropping the dependency for now. I don't _think_ the file dialog works without having portals setup; Other Rust software packages (Zed) don't seem to have fallbacks either.

normalcea commented:
The only other thing I'm worried about for this package is the fact that it's still stuck on an older openh264 version (2.4.1) and the issue has been there for months now. 

I really do wish we could remove the openh264 override since it doesn't feel good shipping software with known [vulnerabilities in them](https://github.com/cisco/openh264/security/advisories/GHSA-m99q-5j7x-7m9x). From my understanding this isn't too serious of an issue since ruffle is geared towards running older, unmaintained flash files rather than say something like Firefox.

normalcea commented:
I replaced `binaryWrapper` with `autoPatchElfHook` which led me to add the required `libgcc` dependency to `buildInputs`  I also removed the extraneous `postPatch` attribute override in `openh264-241`.

jchv reviewed (approved):
Diff seems fine at a glance. Tested on amd64 Linux+X11, amd64 Linux+Wayland and aarch64 Darwin and seems to work fine. No particular notes.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 407724`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ruffle</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ruffle</li>
  </ul>
</details>

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/2386
